Psychtoolbox/PsychHardware/EyelinkToolbox/EyelinkBasic/PsychEyelinkDispatchCallback.m

```diff
@@ -129,326 +161,419 @@
+                    sn = Screen('WindowScreenNumber', eyewin);
+                    offscreen = Screen('OpenOffscreenWindow', sn, el.backgroundcolour, [], [], 32);
+                    Screen(offscreen,'TextFont',el.imgtitlefont);
+                    Screen(offscreen,'TextSize',el.imgtitlefontsize);
+                    Screen('DrawText', offscreen, imgtitle, width/2-dw/2, heigth/2+dh/2+h2, el.imgtitlecolour);
+                    Screen('SelectStereoDrawBuffer', eyewin, it); % select left eye window
+                    Screen('DrawTexture',eyewin,offscreen,  [width/2-dw/2 heigth/2+dh/2+h2 width/2-dw/2+500 heigth/2+dh/2+h2+500], [width/2-dw/2 heigth/2+dh/2+h2 width/2-dw/2+500 heigth/2+dh/2+h2+500]);
+                    Screen('Close',offscreen);
+                    
+                    lasttitle = imgtitle;
+                end
+            catch %myerr
+                fprintf('EyelinkDrawCameraImage:error \n');
+                disp(psychlasterror);
             end
+        end
+    end
+
+
+
 
-            sn = Screen('WindowScreenNumber', eyewin);
-            offscreen = Screen('OpenOffscreenWindow', sn, el.backgroundcolour, [], [], 32);
-            Screen(offscreen,'TextFont',el.imgtitlefont);
-            Screen(offscreen,'TextSize',el.imgtitlefontsize);
-            Screen('DrawText', offscreen, imgtitle, width/2-dw/2, heigth/2+dh/2+h2, el.imgtitlecolour);
-            Screen('DrawTexture',eyewin,offscreen,  [width/2-dw/2 heigth/2+dh/2+h2 width/2-dw/2+500 heigth/2+dh/2+h2+500], [width/2-dw/2 heigth/2+dh/2+h2 width/2-dw/2+500 heigth/2+dh/2+h2+500]);
-            Screen('Close',offscreen);
-            newImage = 0;
+
+    function EyelinkDrawCalibrationTarget(eyewin, el, calxy)
+        if el.winInfo.StereoMode ~= 0
+            drawScreens = 2; % stereoscopic drawing
+        else
+            drawScreens = 1; % non-stereoscopic drawing
+        end
+
+        [width, heigth]=Screen('WindowSize', eyewin);
+
+        for it = 0:drawScreens-1
+            Screen('SelectStereoDrawBuffer', eyewin, it); % select eye window
+            switch el.calTargetType
+                case 'video'
+                    if( ~isempty(el.calAnimationTargetFilename) && ~isempty(eyelinkanimationtarget))
+                        rect=CenterRectOnPoint([0 0 eyelinkanimationtarget.imgw eyelinkanimationtarget.imgh], calxy(1), calxy(2));
+                        if it == 0
+                            tex=Screen('GetMovieImage', eyewin, eyelinkanimationtarget.movie, el.calAnimationWaitTex);
+                        end
+                        if(tex>0)
+                            Screen('DrawTexture', eyewin , tex, [], rect, [], 0);
+                            if drawScreens == 1 || (drawScreens == 2 && it == 1)
+                                Screen('Flip', eyewin);
+                            end
+                        end
+                        if it == drawScreens-1 && tex > 0
```

Comment:
This close texture call can be just put after the 'flip' call above, no need for an extra if-blah-end.

Reply:
resolved in 9e17d4f50104aab97855289e617b8627d7442617